broadinstitute/pearl
# [JN-325] basic admin visibility of study metrics

This adds metrics functionality to the admin tool for simple visualization of enrollment, consent, etc..   (Don't worry, *most* of the 5,700 added lines in this PR are due to package.lock changes from plotly)
![image](https://github.com/broadinstitute/pearl/assets/2800795/b4a093cb-3047-43a6-8153-fc1a62ca7629)

To support testing this, this also adds rudimentary capability for populating users "in the past" via specifying a `submittedHoursAgo` attribute on some populate models.  

Lots of stuff that could be refined as we find out what's useful, but this should be enough for OurHealth in the short term.

TO TEST
1. restart ApiAdminApp
2. repopulate OurHealth
3. run `npm install` from ui-admin (and note the new note in the README if the install fails)
4. go to https://localhost:3000/ourhealth/studies/ourheart/env/sandbox/metrics
5. confirm it looks like the screenshot above.
6. click "Copy raw data" for the top graph, and paste into a google sheet.  You should get something like the data below:
![image](https://github.com/broadinstitute/pearl/assets/2800795/1c9329ab-eae1-44a8-bdd4-41877a0e2925)
